modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/util/DDMIndexerImpl.java

```diff
@@ -490,11 +491,21 @@ protected void addFieldValueRequiredTerm(
 			StringBundler.concat(
 				DDM_FIELD_ARRAY, StringPool.PERIOD, DDM_FIELD_NAME),
 			ddmStructureFieldName);
-		booleanQuery.addRequiredTerm(
-			StringBundler.concat(
-				DDM_FIELD_ARRAY, StringPool.PERIOD,
-				getValueFieldName(indexType, locale)),
-			StringPool.QUOTE + ddmStructureFieldValue + StringPool.QUOTE);
+
+		if (localizableField) {
+			booleanQuery.addRequiredTerm(
+				StringBundler.concat(
+					DDM_FIELD_ARRAY, StringPool.PERIOD,
+					getValueFieldName(indexType, locale)),
+				StringPool.QUOTE + ddmStructureFieldValue + StringPool.QUOTE);
+		}
+		else {
+			booleanQuery.addRequiredTerm(
+				StringBundler.concat(
+					DDM_FIELD_ARRAY, StringPool.PERIOD,
+					getValueFieldName(indexType)),
+				StringPool.QUOTE + ddmStructureFieldValue + StringPool.QUOTE);
+		}
```

Comment:
@davidtellolr, is it possible to reuse some code by doing something like:

```
if (!localizableField) {
    locale = null;
}

booleanQuery.addRequiredTerm(
	StringBundler.concat(
		DDM_FIELD_ARRAY, StringPool.PERIOD,
		getValueFieldName(indexType, locale)),
	StringPool.QUOTE + ddmStructureFieldValue + StringPool.QUOTE);
```

Reply:
Hi @carolmariaabb ,

I'll add your suggestions. Do you prefer these changes in a new PR with a single commit? I think that it will be more clear.

Thanks.

Reply:
I agree with you, @davidtellolr. A single commit would be better in this case.